configure

```diff
@@ -6223,9 +6263,13 @@ esac
 
 else
 
-  case $ac_sys_system in #(
-  Emscripten) :
-    EXEEXT=.wasm ;; #(
+  case $ac_sys_system/$ac_sys_emscripten_target in #(
+  Emscripten/browser) :
+    EXEEXT=.html ;; #(
+  Emscripten/node) :
+    EXEEXT=.js ;; #(
+  wasi/*) :
+    EXEEXT=.js ;; #(
```

Comment:
I think this should be .wasm, since WASI doesn't require JS shims for system APIs.